File: Dante1/Module1.07/Zadanie7.12/main.c
#include <stdio.h>

int is_prime(int a);

int is_twin(int a, int b)
{
    return is_prime(a) && is_prime(b) && ((a - b == 2) || (b - a) == 2);
}

int main()
{
    int a;
    printf("Enter: ");

    int check1 = scanf("%i",&a);
    if(check1 != 1)
    {
        printf("\nIncorrect input\n");
        return 1;
    }

    int b;
    printf("\nEnter: ");

    int check2 = scanf("%i",&b);
    if(check2 != 1)
    {
        printf("\nIncorrect input\n");
        return 1;
    }

    if(b + 2 <= a)
    {
        printf("\nIncorrect input\n");
        return 1;
    }

    printf("\n");
    int pointer = 0;
    for(int i = a; i <= b - 2; i++)
    {
        if(is_twin(i, i + 2))
        {
            printf("%i, %i\n",i, i + 2);
            pointer = 1;
        }
    }
    if(pointer == 0)
    {
        printf("\nNothing to show\n");
    }

    return 0;
}

int is_prime(int a)
{
    int counter = 0;
    if(a == 0 || a == 1 || a < 0)
    {
        return 0;
    }
    for(int i = 1; i <= a; i++)
    {
        if(a % i == 0)
        {
            counter++;
        }
    }
    if(counter <= 2)
    {
        return 1;
    }
    return 0;
}

File: Dante1/Module1.07/Zadanie7.12/main.rs
fn is_prime(a: i32) -> i32 {
    let mut counter = 0;
    if a <= 1 {
        return 0;
    }

    for i in 1..=a {
        if a % i == 0 {
            counter += 1;
        }
    }

    if counter <= 2 {
        return 1;
    }
    0
}

fn is_twin(a: i32, b: i32) -> bool {
    is_prime(a) == 1 && is_prime(b) == 1 && ((a - b == 2) || (b - a) == 2)
}

fn main() {
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("err");
    let mut num1: i32 = input.trim().parse().expect("err");

    input.clear();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("err");
    let mut num2: i32 = input.trim().parse().expect("err");

    if num1 >= num2 + 2 {
        panic!("err");
    }

    let mut counter: usize = 0;
    for i in num1..=num2 - 2 {
        if is_twin(i, i + 2) {
            println!("{} {}", i, i + 2);
            counter += 1;
        }
    }

    if counter == 0 {
        println!("Nothing to show");
    }
}
